mobile/src/common/radio-buttons-group.component.tsx

```diff
@@ -0,0 +1,154 @@
+                onPress={() => {
+                    if (!this.props.disabled) {
+                        this.props.onPress(this.props.label);
+                    }
+                }}>
+                <View
+                    style={[
+                        radioButtonsGroupStyles.radioButtonBorder,
+                        {
+                            borderColor: this.props.color,
+                            width: this.props.size,
+                            height: this.props.size,
+                            borderRadius: this.props.size / 2,
+                            alignSelf: 'center'
+                        },
+                    ]}>
+                    {this.props.selected &&
+                    <View
+                        style={{
+                            backgroundColor: this.props.color,
+                            width: this.props.size / 2,
+                            height: this.props.size / 2,
+                            borderRadius: this.props.size / 2,
+                        }}
+                    />}
+                </View>
+                <StyledText style={textStyle}>{this.props.label}</StyledText>
+            </TouchableOpacity>
+        );
+    }
+}
+
+interface RadioGroupProps {
+    flexDirection?: string;
+    radioButtons: RadioButton[];
+    onPress: (data: RadioButton[]) => void;
+}
+
+interface RadioGroupState {
+    radioButtons: RadioButton[];
+}
+
+export class RadioGroup extends React.Component<RadioGroupProps, RadioGroupState> {
+
+    constructor(props: RadioGroupProps) {
+        super(props);
+
+        this.state = {
+            radioButtons: this.validate(this.props.radioButtons),
+        };
+    }
+
+    public render() {
+        return (
+            <View style={radioButtonsGroupStyles.radioButtonContainer}>
+                <View style={{ flexDirection: this.props.flexDirection } as ViewStyle}>
+                    {this.state.radioButtons.map(button => (
+                        <RadioButtonComponent
+                            key={button.label}
+                            selectionIndex={button.selectionIndex}
+                            label={button.label}
+                            selected={button.selected}
+                            color={button.color}
+                            size={button.size}
+                            disabled={button.disabled}
+                            layout={button.layout}
+                            labelStyle={button.labelStyle}
+                            onPress={this.onPress}
+                        />
+                    ))}
+                </View>
+            </View>
+        );
+    }
+
+    private validate(buttons: RadioButton[]) {
+        let selected = false; // Variable to check if "selected: true" for more than one button.
+        buttons.map(button => {
+            button.color = button.color ? button.color : '#444';
```

Comment:
props shouldn't be mutated

Reply:
Not anymore. 3ffda78b754b078f3ac42b00401ac6da16609321